### ISSUE TITLE
Expose 4-byte ASN support on the config

### ISSUE THREAD

**Issue by johananl** [metallb/metallb#734]
https://github.com/metallb/metallb/pull/491 introduced support for disabling 4-byte ASNs so that MetalLB could be used with routers which don't support RFC 4893. Looks like this functionality has never been exposed in the configuration, which means it is always off - see https://github.com/metallb/metallb/pull/491#issuecomment-702341273.

We should look into adding a peer configuration field which would control whether 4-byte ASNs should be used for that peer.

A possible UX:

```yaml
peers:
- peer-address: 10.0.0.1
  peer-asn: 64501
  my-asn: 64500
  2-byte-asn: true
```

We can then add a `fbasn` boolean argument here...

https://github.com/metallb/metallb/blob/8f2f56213d89765c6c2e541c87eb5fdf13c3e6f6/internal/bgp/bgp.go#L354

...and populate it here:

https://github.com/metallb/metallb/blob/8f2f56213d89765c6c2e541c87eb5fdf13c3e6f6/speaker/bgp_controller.go#L187

In any case the default behavior should be the current state, namely 4-byte ASNs turned **on**.

**Comment by q3k** [comment]
I'm the original author of #491.

My original change checks what the other side supports four-byte ASNs (based on OPEN), and makes decision whether to use them or not depending on that. There generally should be no need to have to configure this - the point of BGP capabilities is that they are only enabled if both sides support them, and no other router I've ever configured needs to have this set explicitly.

For what it's worth, we're running this in production against a device that does not support RFC4893, so at even in practice this seems to just work, too.

**Comment by q3k** [comment]
If anything, #491 could be extended to full RFC 4893 compliance by using AS_TRANS in AS_PATH to establish sessions with non-4893-compliant peers, instead of just failing altogether (as it currently does now, if it has a 4-byte ASN set and talks to a non-4893-compliant peer).

**Comment by johananl** [comment]
Thanks @q3k, I've definitely overlooked this behavior (and wasn't involved with MetalLB when this functionality got merged :-)).

I'll close this then.

**Comment by hibby50** [comment]
@q3k I have confirmed with a packet capture that our switches are not signifying that they support 4 byte ASN in their open messages, but our update messages are still being sent with 4 byte ASNs causing the exact issue outlined in #491 Is there anything else you are doing to make the speakers send 2 byte ASNs?

**Comment by q3k** [comment]
Could you post the packet capture here, and your metallb configuration? The only thing that comes to mind that your metallb is configured with a 4-byte ASN instead of one that fits in 2 bytes. Although that should be throwing an error instead.

Our metallb configuration is [public](https://cs.hackerspace.pl/hscloud@e401735fdd241b25dac4cb82d828dcfa6f84b198/-/blob/cluster/kube/k0.libsonnet#L21-L49) if you want to refer to it.

**Comment by hibby50** [comment]
@q3k This ended up being a mistake on my part. We have an MLAG switch configuration, and after dumbing our setup down to only one worker and one switch and realizing it still wasn't falling back to 2 byte, I looked more closely at our configs and realized we were pulling the master tag on docker hub which does not include your pull req. I pulled a different tag and your code is working as intended on our MLAG setup. Thank you for your interest in helping as well as the initial code, sorry for any confusion :)
